examples/testdata/testframework/impl/framework.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2023 Jetpack Technologies Inc and contributors. All rights reserved.
+// Use of this source code is governed by the license in the LICENSE file.
+package impl
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os/exec"
+
+	"github.com/pkg/errors"
+	"go.jetpack.io/devbox/internal/impl/shellcmd"
+)
+
+type DevboxJson struct {
+	Packages []string `cue:"[...string]" json:"packages"`
+	Shell    struct {
+		InitHook shellcmd.Commands             `json:"init_hook,omitempty"`
+		Scripts  map[string]*shellcmd.Commands `json:"scripts,omitempty"`
+	} `json:"shell,omitempty"`
+
+	Nixpkgs struct {
+		Commit string `json:"commit,omitempty"`
+	} `json:"nixpkgs,omitempty"`
+}
+
+type TestDevbox struct {
+	devboxJsonPath string
+}
+
+func (td *TestDevbox) Info(pkg string, markdown bool) (string, error) {
+	cmd := exec.Command("devbox", "info", pkg)
```

Comment:
I may be missing something, but doesn't this rely on the `devbox` binary being compiled and being present in the path. How or where is that done when executing the golang tests....?

Reply:
I'm wondering if we can use the approach taken in this article?
https://gianarb.it/blog/golang-mockmania-cli-command-with-cobra

excerpt:
```
func NewRootCmd() *cobra.Command {
	cmd := &cobra.Command{
		Use:   "hugo",
		Short: "Hugo is a very fast static site generator",
		RunE: func(cmd *cobra.Command, args []string) error {
			fmt.Fprintf(cmd.OutOrStdout(), in)
			return nil
		},
	}
	cmd.Flags().StringVar(&in, "in", "", "This is a very important input.")
	return cmd
}

func Test_ExecuteCommand(t *testing.T) {
	cmd := NewRootCmd()
	b := bytes.NewBufferString("")
	cmd.SetOut(b)
	cmd.SetArgs([]string{"--in", "testisawesome"})
	cmd.Execute()
	out, err := ioutil.ReadAll(b)
	if err != nil {
		t.Fatal(err)
	}
	if string(out) != "testisawesome" {
		t.Fatalf("expected \"%s\" got \"%s\"", "testisawesome", string(out))
	}
}
```

so that way, we can compile the command and also inspect the stdout and stderr?

Reply:
tI changed the testing method from `exec.Command` to this. The only issue is that now, I can't read the shell output from the `Run` command. For example, if I have a script in my devbox.json that says: `echo 'hello'`, the `hello` will neither appear in stdout nor stderr of the command.
Not a big deal for now, we can just verify there was no error running `devbox run` command.